packages/happn-3/lib/services/protocol/service.js

```diff
@@ -223,20 +212,48 @@ function __processSinglePath(transformed, callback) {
       });
     }
 
+    if (authorized.request.action === 'remove')
+      return this.happn.services.data.processRemove(authorized, (e, publication) => {
+        if (e) return callback(e);
+        if (publication.request.options && publication.request.options.noPublish)
```

Comment:
may as well tidy this up with: if ( publication.request?.options?.noPublish)

Reply:
Which version of node are we targeting? Guessing > 16

Reply:
Fixed.